test/dotnet-new.IntegrationTests/DotnetNewTestTemplatesTests.cs

```diff
@@ -0,0 +1,259 @@
+        }
+
+        static DotnetNewTestTemplatesTests()
+        {
+            string templatePackagePath = Path.Combine(
+                RepoTemplatePackages,
+                "Microsoft.DotNet.Common.ProjectTemplates.10.0",
+                "content");
+
+            var dummyLog = new NullTestOutputHelper();
+
+            new DotnetNewCommand(dummyLog, "uninstall", templatePackagePath)
+                   .WithCustomHive(CreateTemporaryFolder(folderName: "Home"))
+                   .WithWorkingDirectory(CreateTemporaryFolder())
+                   .Execute();
+
+            new DotnetNewCommand(dummyLog, "install", templatePackagePath)
+                .WithCustomHive(CreateTemporaryFolder(folderName: "Home"))
+                .WithWorkingDirectory(CreateTemporaryFolder())
+                .Execute()
+                .Should().ExitWith(0);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTemplateItemsToTest))]
+        public void ItemTemplate_CanBeInstalledAndTestArePassing(string targetFramework, string projectTemplate, string itemTemplate, string language)
+        {
+            string testProjectName = GenerateTestProjectName();
+            string outputDirectory = CreateTemporaryFolder(folderName: "Home");
+            string workingDirectory = CreateTemporaryFolder();
+
+            // Create new test project: dotnet new <projectTemplate> -n <testProjectName> -f <targetFramework> -lang <language>
+            string args = $"{projectTemplate} -n {testProjectName} -f {targetFramework} -lang {language} -o {outputDirectory}";
+            new DotnetNewCommand(_log, args)
+             .WithCustomHive(outputDirectory).WithRawArguments()
+             .WithWorkingDirectory(workingDirectory)
+             .Execute()
+             .Should().ExitWith(0);
+
+            var itemName = "test";
+
+            // Add test item to test project: dotnet new <itemTemplate> -n <test> -lang <language> -o <testProjectName>
+            new DotnetNewCommand(_log, $"{itemTemplate} -n {itemName} -lang {language} -o {outputDirectory}")
+                .WithCustomHive(outputDirectory).WithRawArguments()
+                .WithWorkingDirectory(workingDirectory)
+                .Execute()
+                .Should().ExitWith(0);
+
+            if (language == Languages.FSharp)
+            {
+                // f# projects don't include all files by default, so the file is created
+                // but the project ignores it until you manually add it into the project
+                // in the right order
+                AddItemToFsproj(itemName, outputDirectory, testProjectName);
+            }
+
+            var result = new DotnetTestCommand(_log, false)
+                .WithWorkingDirectory(outputDirectory)
+                .Execute(outputDirectory);
+
+            result.Should().ExitWith(0);
+
+            result.StdOut.Should().Contain("Passed!");
+            result.StdOut.Should().MatchRegex(@"Passed:\s*2");
```

Comment:
There are two tests that run?

Reply:
Probably worth adding some comment in the code @mariam-abdulla .